### PR TITLE
Edb data refactoring 2.0

### PR DESCRIPTION
**Pull request by svandenb-dev**

### CODE REVIEW

**Review by maxcapodi78 — APPROVED** [review]

File: pyaedt/edb_core/edb_data/components_data.py [hunk 1]
```diff
@@ -47,6 +47,12 @@ def part_name(self, name):
 
     @property
     def type(self):
+        """Retryve the Component definition type.
```
Comment (PipKat) [inline pyaedt/edb_core/edb_data/components_data.py:50]:
```suggestion
        """Retrieve the component definition type.
```

File: pyaedt/edb_core/edb_data/simulation_configuration.py [hunk 2]
```diff
@@ -14,17 +14,18 @@
...
 except ImportError:
     if os.name != "posix":
         warnings.warn("This module requires pythonnet.")
 
 
 class SimulationConfiguration(object):
-    """Parses an ASCII simulation configuration file, which supports all types of inputs
+    """ASCII simulation configuration file parser, which supports all types of inputs
     for setting up and automating any kind of SI or PI simulation with HFSS 3D Layout
     or Siwave. If fields are omitted, default values are applied. This class can be instantiated directly from
```
Comment (PipKat) [inline pyaedt/edb_core/edb_data/simulation_configuration.py:27]:
```suggestion
    """Provides an ASCII simulation configuration file parser.
    
    This parser supports all types of inputs for setting up and automating any kind
    of SI or PI simulation with HFSS 3D Layout or Siwave. If fields are omitted, default
    values are applied. This class can be instantiated directly from
```

Comment (PipKat) [reply pyaedt/edb_core/edb_data/simulation_configuration.py:27]: Alex says that the descriptions of classes and methods have to be kept short so they can fully display in tables and summary pages. I've reformatted the content to do this.

File: pyaedt/edb_core/edb_data/simulation_configuration.py [hunk 3]
```diff
@@ -2330,3 +2331,350 @@ def add_rlc(
             return True
         except:  # pragma: no cover
             return False
+
+
+class SiwaveDCSetupTemplate(object):
+    """Siwave DC Settings Data Class.
```
Comment (PipKat) [inline pyaedt/edb_core/edb_data/simulation_configuration.py:2337]:
```suggestion
    """Provides Siwave DC settings data.
```

File: pyaedt/edb_core/edb_data/simulation_configuration.py [hunk 4]
```diff
@@ -2330,3 +2331,350 @@ def add_rlc(
...
+
+    This class contains all the settings for a Siwave DC Analysis and
```
Comment (PipKat) [inline pyaedt/edb_core/edb_data/simulation_configuration.py:2339]:
```suggestion
    This class contains all the settings for a Siwave DC analysis and
```

File: pyaedt/edb_core/edb_data/simulation_configuration.py (hunk 2 above)
Comment (maxcapodi78) [reply pyaedt/edb_core/edb_data/simulation_configuration.py]: Thanks. Good point

**Pull request merged**